docs/artifacts.md

```diff
@@ -0,0 +1,97 @@
+Artifact Versioning and Tagging (1.0.0)
+=======================================
+
+Artifact Name Composition
+-------------------------
+
+### Filenames
+
+`<--artifact_name-->@<--semantic_version-->.<--file_extension-->`
+
+Example:
+
+`electron-bootloader@1.0.1+lto.debug.jtag.bin`
+
+#### Filename Composition
+
+- Artifact Name
+  - (Platfrom Family Name (_none_) | Platform Name (`electron`))
+  - Module Name (`bootloader`)
+- Semantic Version
+  - Major Version (`1`)
+  - Minor Version (`0`)
+  - Patch Version (`1`)
+  - [_OPTIONAL_] Prerelease Tag (_none_)
+  - [_OPTIONAL_] Build Metadata 1,2,...,n (`lto`, `debug`, `jtag`)
```

Comment:
Currently, the document describes the handling of boolean build metadata. If we decide to start representing the monolithic flag in our build metadata, then it will naturally fit here.